extension/nine-patch/src/commonMain/kotlin/com/seiko/imageloader/util/NinePatchPainter.kt

```diff
@@ -0,0 +1,221 @@
+package com.seiko.imageloader.util
+
+import androidx.compose.ui.geometry.Size
+import androidx.compose.ui.graphics.ColorFilter
+import androidx.compose.ui.graphics.ImageBitmap
+import androidx.compose.ui.graphics.drawscope.DrawScope
+import androidx.compose.ui.graphics.painter.Painter
+import androidx.compose.ui.unit.IntOffset
+import androidx.compose.ui.unit.IntSize
+import com.seiko.imageloader.model.NinePatchData
+import kotlin.math.max
+import kotlin.math.min
+import kotlin.math.roundToInt
+
+internal class NinePatchPainter(
+    private val image: ImageBitmap,
+    private val ninePatchData: NinePatchData,
+) : Painter() {
+
+    private val scale get() = ninePatchData.scale
+    private val skipPadding get() = ninePatchData.skipPadding
+    private val maxFactor get() = ninePatchData.maxFactor
+    private val filterQuality get() = ninePatchData.filterQuality
+
+    private val centerWidth = ninePatchData.right - ninePatchData.left
```

Comment:
in this place there may be a case when size = 0
if the original size does not contain pixels to draw, then the draw function will not draw anything in the center.